docs/howto/openstack/cinder/retype-volumes.md

```diff
@@ -0,0 +1,106 @@
+| ID           | Name    | Status | Size | Type        | Bootable | Attached to  | Properties |
++--------------+---------+--------+------+-------------+----------+--------------+------------+
+| d2a22868-a13 | testvol | in-use |    2 | __DEFAULT__ | false    | Attached to  |            |
+| 3-40a1-b1a6- |         |        |      |             |          | testsrv on   |            |
+| 0cbae3feaf8d |         |        |      |             |          | /dev/vdb     |            |
++--------------+---------+--------+------+-------------+----------+--------------+------------+
+```
+
+As you can see, this volume's status is `in-use`, meaning it is currently attached to a server, and the volume's type is `__DEFAULT___`.
+
+
+## Detaching the volume
+
+You cannot retype a volume while it is attached to a server.
+You must thus detach it first.
+It is safest to do this while the server is shut down:
+
+```console
+$ openstack server stop testsrv
+
+$ openstack server list -c Name -c Status
++---------+---------+
+| Name    | Status  |
++---------+---------+
+| testsrv | SHUTOFF |
++---------+---------+
+```
+
+Once your server is in the `SHUTOFF` state, you can safely proceed to detaching the volume.
+This will change the volume's status from `in-use` to `available`.
+
+```console
+$ openstack server remove volume testsrv testvol
+
+$ openstack volume list
++--------------------------------------+---------+-----------+------+-------------+
+| ID                                   | Name    | Status    | Size | Attached to |
++--------------------------------------+---------+-----------+------+-------------+
+| d2a22868-a133-40a1-b1a6-0cbae3feaf8d | testvol | available |    2 |             |
++--------------------------------------+---------+-----------+------+-------------+
+```
+
+## Retyping the volume
+
+With the volume safely detached, you can now change its volume type.
+If you want to change the volume type from `__DEFAULT__` to `ceph_hdd`, you would proceed as follows:
+
+```console
+$ openstack volume set --type ceph_hdd --retype-policy on-demand testvol
+
+$ openstack volume list --long
++---------------+---------+-----------+------+----------+----------+-------------+------------+
+| ID            | Name    | Status    | Size | Type     | Bootable | Attached to | Properties |
++---------------+---------+-----------+------+----------+----------+-------------+------------+
+| d2a22868-a133 | testvol | available |    2 | ceph_hdd | false    |             |            |
+| -40a1-b1a6-0c |         |           |      |          |          |             |            |
+| bae3feaf8d    |         |           |      |          |          |             |            |
++---------------+---------+-----------+------+----------+----------+-------------+------------+
+```
+
+It is at this stage that the actual data migration occurs.
+As a result, this step might take a significant amount of time.
```

Comment:
@noonedeadpunk, in this section we're still missing the bit where the user should check on whether the migration has completed (i.e. where I don't know if they're even able to, without admin privileges).